src/libpcp_web/src/query.c

```diff
@@ -2623,6 +2627,69 @@ series_calculate_rate(node_t *np)
     }
 }
 
+/*
+ * Compare and pick the max instance value(s) among samples.
+ */
+static void
+series_calculate_time_domain_max(node_t *np)
+{
+    seriesQueryBaton	*baton = (seriesQueryBaton *)np->baton;
+    unsigned int	n_series, n_samples, n_instances, i, j, k;
+    double		max_data, data;
+    int			max_pointer;
+    sds			msg;
+
+    n_series = np->left->value_set.num_series;
+    np->value_set.num_series = n_series;
+    np->value_set.series_values = (series_sample_set_t *)calloc(n_series, sizeof(series_sample_set_t));
+    for (i = 0; i < n_series; i++) {
+		n_samples = np->left->value_set.series_values[i].num_samples;
+		if (n_samples > 0){
+			np->value_set.series_values[i].num_samples = n_samples;
+			np->value_set.series_values[i].series_sample = (series_instance_set_t *)calloc(n_samples, sizeof(series_instance_set_t));
+			n_instances = np->left->value_set.series_values[i].series_sample[0].num_instances;
```

Comment:
Did you find the cause of the memory allocation / valgrind issue from the other day?  (was it n_samples == 0?)